File: test_locales.js
const fs = require('fs');
const content = fs.readFileSync('c:/Users/LENOVO/Downloads/skripsi-tes - Copy (2)/js/locales.js', 'utf8');

const enBlockMatch = content.match(/en:\s*{([\s\S]+?)},\s*id:/);
const idBlockMatch = content.match(/id:\s*{([\s\S]+?)}\s*};/);

const enBlock = enBlockMatch[1];
const idBlock = idBlockMatch[1];

function extractKeys(text) {
  const keys = new Set();
  const lines = text.split('\n');
  for (let line of lines) {
    // remove single line comments
    let cleanLine = line.split('//')[0];
    const matches = [...cleanLine.matchAll(/(?:^|[\s,])([a-zA-Z0-9_]+)\s*:\s*["']/g)];
    for (let m of matches) {
      keys.add(m[1]);
    }
  }
  return keys;
}

const enKeys = Array.from(extractKeys(enBlock));
const idKeys = Array.from(extractKeys(idBlock));

const missingInId = enKeys.filter(k => !idKeys.includes(k));
const missingInEn = idKeys.filter(k => !enKeys.includes(k));

console.log('Missing in ID:', missingInId.sort());
console.log('Missing in EN:', missingInEn.sort());

File: test_locales.py
import re

with open("c:/Users/LENOVO/Downloads/skripsi-tes - Copy (2)/js/locales.js", "r", encoding="utf-8") as f:
    content = f.read()

en_block = re.search(r'en:\s*{([\s\S]+?)},\s*id:', content).group(1)
id_block = re.search(r'id:\s*{([\s\S]+?)}\s*};', content).group(1)

def extract_keys(text):
    # Match any alphanumeric/underscore word followed by a colon and quote
    matches = re.findall(r'([a-zA-Z0-9_]+)\s*:\s*["\']', text)
    return set(matches)

en_keys = extract_keys(en_block)
id_keys = extract_keys(id_block)

print(f"EN Keys: {len(en_keys)}")
print(f"ID Keys: {len(id_keys)}")

missing_in_id = en_keys - id_keys
missing_in_en = id_keys - en_keys

if missing_in_id:
    print("Missing in ID:", sorted(list(missing_in_id)))
if missing_in_en:
    print("Missing in EN:", sorted(list(missing_in_en)))

if not missing_in_id and not missing_in_en:
    print("Translations are fully synced!")
